tests/test_visitors/test_ast/test_loops/test_comprehensions/test_unsafe_generator_expression.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from wemake_python_styleguide.violations.consistency import (
+    UnsafeGeneratorExpressionViolation,
+)
+from wemake_python_styleguide.visitors.ast.statements import (
+    UnsafeGeneratorExpressionVisitor,
+)
+
+unsafe_function = """
+first_value = 1
+expression = (first_value * index for index in range(5))
+first_value = 2
```

Comment:
What about changing `index`? Let's test that as well.

Reply:
I believe that the index cannot be changed as the variable is load within the generator expression. Even though, I'll include a test